src/sentry/integrations/discord/client.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+from requests import PreparedRequest
+
+from sentry import options
+from sentry.services.hybrid_cloud.util import control_silo_function
+from sentry.shared_integrations.client.proxy import IntegrationProxyClient
+from sentry.shared_integrations.exceptions import IntegrationError
+from sentry.shared_integrations.exceptions.base import ApiError
+from sentry.utils.json import JSONData
+
+
+class DiscordClient(IntegrationProxyClient):
+    integration_name: str = "discord"
+    base_url: str = "https://discord.com/api/v10"
+
+    # https://discord.com/developers/docs/resources/guild#get-guild
+    GET_GUILD_URL = "/guilds/%s"
+
+    def __init__(
+        self,
+        org_integration_id: int | None = None,
+        logging_context: JSONData | None = None,
+    ):
+        self.application_id = options.get("discord.application-id")
+        self.bot_token = options.get("discord.bot-token")
```

Comment:
@spalmurray you'll need to register these options in https://github.com/getsentry/sentry/blob/master/src/sentry/options/defaults.py